src/css.js

```diff
@@ -257,7 +264,11 @@ jQuery.extend( {
 			if ( !hooks || !( "set" in hooks ) ||
 				( value = hooks.set( elem, value, extra ) ) !== undefined ) {
 
-				style[ name ] = value;
+				if ( isCustomProp ) {
```

Comment:
I originally planned to remove it as discussed in #3199 to fix https://bugs.jquery.com/ticket/14394 but `CSSStyleDeclaration#setProperty` requires styles in the kebab-case format so whatever we decide about that I'd like it to land separately.